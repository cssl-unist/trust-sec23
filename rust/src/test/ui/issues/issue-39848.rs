macro_rules! get_opt {
    ($tgt:expr, $field:ident) => {
        if $tgt.has_$field() {} //~ ERROR expected `{`, found `foo`
    }
}

fn main() {
    get_opt!(bar, foo);
}
