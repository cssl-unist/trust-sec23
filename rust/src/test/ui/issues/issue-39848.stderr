error: expected `{`, found `foo`
  --> $DIR/issue-39848.rs:3:21
   |
LL |         if $tgt.has_$field() {}
   |         --          ^^^^^^--
   |         |           |
   |         |           expected `{`
   |         |           help: try placing this code inside a block: `{ $field() }`
   |         this `if` expression has a condition, but no block
...
LL |     get_opt!(bar, foo);
   |     ------------------- in this macro invocation
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

